modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SortConfigurationOptions.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayForm, {ClayInput, ClayToggle} from '@clayui/form';
+import React, {useState} from 'react';
+
+import FieldList, {ORDERS} from './FieldList';
+
+/**
+ * Adds the possible order symbol (+/-) at the end of the indexed field name.
+ * @param {Array} fields Field array of objects with fieldName, label, and order.
+ * @return {Array} Field array of objects with fieldName (has the order symbol)
+ * and label.
+ */
+const addOrdersToFieldNames = (fields) =>
+	fields.map(({field: fieldName, label, order}) => ({
+		field: `${fieldName}${order === ORDERS.DESC.value ? '-' : '+'}`,
+		label,
+	}));
+
+/**
+ * Uses the indexed field name to determine if order is ascending (+) or
+ * descending (-)
+ * @param {string} fieldName The indexed field name with possible order symbol.
+ * @return {string} Order value (ascending or descending).
+ */
+const getOrderFromFieldName = (fieldName) => {
+	return fieldName.slice(-1) === '-' ? ORDERS.DESC.value : ORDERS.ASC.value;
+};
+
+/**
+ * Cleans up the fields array by removing those that have empty indexed
+ * field names.
+ * @param {Array} fields The list of fields.
+ * @return {Array} The cleaned up list of fields.
+ */
+const removeEmptyFields = (fields) =>
+	fields.filter(({field: fieldName}) => fieldName);
+
+/**
+ * Removes the possible order symbol (+/-) at the end of the indexed field
+ * name.
+ * @param {string} fieldName The indexed field name with possible order symbol.
+ * @return {string} The indexed field name without order symbol.
+ */
+const removeOrderFromFieldName = (fieldName) => {
+	return fieldName.slice(-1) === '+' || fieldName.slice(-1) === '-'
+		? fieldName.slice(0, -1)
+		: fieldName;
+};
+
+function SortConfigurationOptions({
+	fieldsInputName = '',
+	fieldsJSONArray = [
+		{field: '', label: 'relevance'},
+		{field: 'modified-', label: 'modified'},
+		{field: 'modified+', label: 'modified-oldest-first'},
+		{field: 'created-', label: 'created'},
+		{field: 'created+', label: 'created-oldest-first'},
+		{field: 'userName', label: 'user'},
+	],
```

Comment:
I don't think we should define these defaults since the configurations should already be set by `SortPortletPreferencesImpl.java` and I'd think that this fallback value would never be used? `[]` should be fine though.

Reply:
Alright! I added `[]` and a small update to prevent error when I was trying it on Storybook.